packages/engine/src/networking/systems/IncomingNetworkSystem.ts

```diff
@@ -117,8 +131,18 @@ export default async function IncomingNetworkSystem(world: World): Promise<Syste
           xrInputSourceComponent.controllerRight.position.fromArray(rightPose)
           xrInputSourceComponent.controllerRight.quaternion.fromArray(rightPose, 3)
         }
+
+        for (const vel of newWorldState.velocities) {
+          if (!Network.instance.networkObjects[vel.networkId]) continue
+          const entity = Network.instance.networkObjects[vel.networkId].entity
+
+          if (hasComponent(entity, VelocityComponent)) {
+            const velC = getComponent(entity, VelocityComponent)
+            velC.velocity.fromArray(vel.velocity === 0 ? [0, 0, 0] : vel.velocity)
```

Comment:
why would velocity ever be 0 here? best to not create an empty object here too, and instead use setScalar(0) to avoid unnecessary GC